src/DataSource/Http.hs

```diff
@@ -109,25 +109,32 @@ listAssignedTickets userId = do
     let url = showURL $ UserAssignedTicketsURL userId
     let req = apiRequest cfg url
 
-    iterateTicketPages req
+    iteratePages req
 
--- | Iterate all the ticket pages and combine into a result.
-iterateTicketPages
-    :: forall m. (MonadIO m, MonadReader Config m)
-    => Request -> m [TicketInfo]
-iterateTicketPages req = do
+listAdminAgents :: forall m. (MonadIO m, MonadReader Config m) => m [User]
+listAdminAgents = do
+    cfg <- ask
+    let url = showURL AgentGroupURL
+    let req = apiRequest cfg url
 
+    iteratePages req
+
+-- | Iterate all the ticket pages and combine into a result.
+iteratePages :: forall m a. (MonadIO m, MonadReader Config m, FromJSON a)
```

Comment:
Perhaps it would be better if we can add some kind of equality constraint on `a` that `a` have to be either `User` or `TicketInfo` but I'm not sure how you can do this.